src/signTx.c

```diff
@@ -349,26 +382,56 @@ static void signTx_handleInit_ui_runStep()
 
 	UI_STEP_BEGIN(ctx->ui_step, this_fn);
 
-	UI_STEP(HANDLE_INIT_STEP_DISPLAY_DETAILS) {
-		char* header =
-		        (ctx->commonTxData.txSigningMode == SIGN_TX_SIGNINGMODE_MULTISIG_TX) ?
-		        "Multisig transaction" :
-		        "New transaction";
+	UI_STEP(HANDLE_INIT_STEP_PROMPT_SIGNINGMODE) {
+		char bodyTxt[SIZEOF("Pool operator") + 1 + SIZEOF("transaction?") + 1];
```

Comment:
I'd consider a bit different approach:
1. allocate a fixed-length buffer (large enough, perhaps 30, even better if you used something corresponding to the screen max line length, but the font is probably not fixed-width)
2. write into the buffer
3. assert that the text written into the buffer does not fill the entire buffer (there is space left even after '\0', i.e. the allocated size is sufficient)
What do you think?

Reply:
But that is what is happening here. bodxTxt is a fixed sized buffer, and snprintf makes sure that we don't overwrite it, including the 0 at the end.

Reply:
My idea was that you not only want to assert that buffer was not oveflowed, but also that all the text that is supposed to be printed is actually printed to the buffer (and that can be assured by requiring some empty space left even after \0) --- we don't want a situation in the future that a tx description is cut off, it could lead to confusion / misunderstanding / loss of funds (that's probably too much of a stretch, but we are purposefully paranoid :) ).

Reply:
I see, that makes sense. Missed that detail in your first comment.